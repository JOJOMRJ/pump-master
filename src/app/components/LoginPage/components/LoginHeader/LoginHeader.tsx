import React from 'react';

export const LoginHeader: React.FC = () => {
  return (
    <div className="text-center mb-4">
      <h2 className="h3 mb-1">Welcome back</h2>
    </div>
  );
};

export default LoginHeader;
