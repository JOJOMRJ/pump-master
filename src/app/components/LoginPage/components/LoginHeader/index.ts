export { LoginHeader } from './LoginHeader';
export { default } from './LoginHeader';
