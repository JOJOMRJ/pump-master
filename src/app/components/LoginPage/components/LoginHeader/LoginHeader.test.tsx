import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { LoginHeader } from './LoginHeader';

describe('LoginHeader', () => {
  it('should render welcome message', () => {
    render(<LoginHeader />);

    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Welcome back');
  });

  it('should have correct Bootstrap classes', () => {
    render(<LoginHeader />);

    const container = screen.getByText('Welcome back').parentElement;
    expect(container).toHaveClass('text-center', 'mb-4');
  });

  it('should render as h2 element with h3 class', () => {
    render(<LoginHeader />);

    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveClass('h3', 'mb-1');
  });

  it('should render without any props', () => {
    expect(() => render(<LoginHeader />)).not.toThrow();
  });

  it('should match snapshot', () => {
    const { container } = render(<LoginHeader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
