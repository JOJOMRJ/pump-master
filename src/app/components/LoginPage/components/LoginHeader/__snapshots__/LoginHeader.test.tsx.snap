// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginHeader > should match snapshot 1`] = `
<div
  class="text-center mb-4"
>
  <h2
    class="h3 mb-1"
  >
    Welcome back
  </h2>
</div>
`;
